src/Database/DatabaseUpdater.php

```diff
@@ -0,0 +1,184 @@
+        if (($newName = $this->table->getName()) != $this->originalTable->getName()) {
+            // Make sure the new name doesn't already exist
+            if (SchemaManager::tableExists($newName)) {
+                throw SchemaException::tableAlreadyExists($newName);
+            }
+        } else {
+            $newName = false;
+        }
+
+        // Rename columns
+        if ($renamedColumnsDiff = $this->getRenamedColumnsDiff()) {
+            SchemaManager::alterTable($renamedColumnsDiff);
+
+            // Refresh original table after renaming the columns
+            $this->originalTable = SchemaManager::listTableDetails($this->tableArr['oldName']);
+        }
+
+        $tableDiff = $this->originalTable->diff($this->table);
+
+        // Add new table name to tableDiff
+        if ($newName) {
+            if (!$tableDiff) {
+                $tableDiff = new TableDiff($this->tableArr['oldName']);
+            }
+
+            $tableDiff->newName = $newName;
+        }
+
+        // Update the table
+        if ($tableDiff) {
+            SchemaManager::alterTable($tableDiff);
+        }
+    }
+
+    /**
+     * Get the table diff to rename columns.
+     *
+     * @return \Doctrine\DBAL\Schema\TableDiff
+     */
+    protected function getRenamedColumnsDiff()
+    {
+        $renamedColumns = $this->getRenamedColumns();
+
+        if (empty($renamedColumns)) {
+            return false;
+        }
+
+        $renamedColumnsDiff = new TableDiff($this->tableArr['oldName']);
+
+        foreach ($renamedColumns as $oldName => $newName) {
+            $renamedColumnsDiff->renamedColumns[$oldName] = $this->table->getColumn($newName);
+        }
+
+        return $renamedColumnsDiff;
+    }
+
+    /**
+     * Get the table diff to rename columns and indexes.
+     *
+     * @return \Doctrine\DBAL\Schema\TableDiff
+     */
+    protected function getRenamedDiff()
+    {
+        $renamedColumns = $this->getRenamedColumns();
+        $renamedIndexes = $this->getRenamedIndexes();
+
+        if (empty($renamedColumns) && empty($renamedIndexes)) {
+            return false;
+        }
+
+        $renamedDiff = new TableDiff($this->tableArr['oldName']);
+
+        foreach ($renamedColumns as $oldName => $newName) {
+            $renamedDiff->renamedColumns[$oldName] = $this->table->getColumn($newName);
+        }
+
+        foreach ($renamedIndexes as $oldName => $newName) {
+            $renamedDiff->renamedIndexes[$oldName] = $this->table->getIndex($newName);
+        }
+
+        return $renamedDiff;
+    }
+
+    /**
+     * Get columns that were renamed.
+     *
+     * @return array
+     */
+    protected function getRenamedColumns()
+    {
+        $renamedColumns = [];
+
+        foreach ($this->tableArr['columns'] as $column) {
+            $oldName = $column['oldName'];
+
+            // make sure this is an existing column and not a new one
+            if ($this->originalTable->hasColumn($oldName)) {
```

Comment:
This `if` statement an the `foreach` could be turned into a collection filter instead for easier reading.

```
collect($this->tableArr['columns'])->filter(function ($column) {
    return $this->originalTable->hasColumn($oldName) && $column['name'] != $column['oldName'];
})->each(function ($column) use (&$renamedColumns) {
    $renamedColumns[$column['oldName']] = $column['name'];
});
```

Thoughts?